packages/@aws-cdk/aws-sns-subscriptions/test/subs.test.ts

```diff
@@ -85,7 +85,42 @@ test('url subscription (unresolved url with protocol)', () => {
           "TopicName": "topicName"
         }
       },
-      "MyTopicUnresolvedUrlBA127FB3": {
+      "MyTopicTokenSubscription141DD1BE2": {
+        "Type": "AWS::SNS::Subscription",
+        "Properties": {
+          "Endpoint": "{{resolve:secretsmanager:my-secret:SecretString:::}}",
+          "Protocol": "https",
+          "TopicArn": { "Ref": "MyTopic86869434" },
+        }
+      }
+    }
+  });
+});
+
+test('url subscription (double unresolved url with protocol)', () => {
+  const secret = SecretValue.secretsManager('my-secret');
+  const url = secret.toString();
+  topic.addSubscription(new subs.UrlSubscription(url, {protocol: sns.SubscriptionProtocol.HTTPS}));
+  topic.addSubscription(new subs.UrlSubscription(url, {protocol: sns.SubscriptionProtocol.HTTPS}));
```

Comment:
Let's use different tokens for these, so we can assert that they resolve to the right tokens.

same for other tests.

Reply:
Updated